src/Lecture1.hs

```diff
@@ -123,4 +142,15 @@ and lower than 6 elements (4, 5, 6, 7, 8 and 9).
 
 🕯 HINT: Use recursion to implement this function.
 -}
-lowerAndGreater n list = error "TODO"
+lowerAndGreater :: Integer -> [Integer] -> String
+lowerAndGreater n list = go n (0 :: Integer) (0 :: Integer) list
```

Comment:
You don't need to pass `n` as an argument to `go` because it's in scope for the local function inside `where`. So you can just use `n` from the function definition without the need to pass it.

Also, you can specify the type of `go` explicitly and you could pass `0` without the need to specify types. And it'll be just:

```haskell
lowerAndGreater n list = go 0 0 list
```

⚠️ Future spoiler: in the future I'll explain eta-reduction. So it's actually possible to reduce "common suffix of arguments" from both sides of `=` and the definition of this function will become even simpler:

```haskell
lowerAndGreater n = go 0 0
```